crates/oxc_transformer/src/es2018/mod.rs

```diff
@@ -0,0 +1,40 @@
+mod object_rest;
+mod object_rest_spread;
+mod object_spread;
+mod options;
+
+pub use object_rest_spread::{ObjectRestSpread, ObjectRestSpreadOptions};
+pub use options::ES2018Options;
+use oxc_ast::ast::*;
+use oxc_traverse::{Traverse, TraverseCtx};
+use std::rc::Rc;
+
+use crate::context::Ctx;
+
+#[allow(dead_code)]
+pub struct ES2018<'a> {
+    ctx: Ctx<'a>,
+    options: ES2018Options,
+
+    // Plugins
+    object_rest_spread: ObjectRestSpread<'a>,
+}
+
+impl<'a> ES2018<'a> {
+    pub fn new(options: ES2018Options, ctx: Ctx<'a>) -> Self {
+        Self {
+            object_rest_spread: ObjectRestSpread::new(
+                options.object_rest_spread.unwrap_or_default(),
+                Rc::clone(&ctx),
+            ),
+            ctx,
+            options,
+        }
+    }
+}
+
+impl<'a> Traverse<'a> for ES2018<'a> {
+    fn enter_expression(&mut self, expr: &mut Expression<'a>, ctx: &mut TraverseCtx<'a>) {
+        self.object_rest_spread.enter_expression(expr, ctx);
```

Comment:
We only run it when this plugin is enabled. You can refer to https://github.com/oxc-project/oxc/blob/d304d6f9734b486f63eb829ab719b85f52d705c2/crates/oxc_transformer/src/es2020/mod.rs#L53-L57

```suggestion
        if self.object_rest_spread.is_some() {
          self.object_rest_spread.enter_expression(expr, ctx);
        }
```

I think that's why so many unrelated test changes were made